core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataReader.java

```diff
@@ -170,6 +170,18 @@ private DataBuffer readChunkInternal() throws IOException {
       return null;
     }
     mPosToRead += buffer.readableBytes();
+    if (mPosToRead - mReadRequest.getOffset() == mReadRequest.getLength()) {
```

Comment:
On second thought; can we do proactive closing in the BlockInStream rather than here? 
That way, you can also consume the buffer before calling close(), and effectively achieve what @yuzhu was asking.